### DISCUSSION TITLE
Can't deploy firebase functions

### DISCUSSION THREAD

**Discussion by BUONJG** [pnpm/pnpm#4079] [Q&A]
Hello,

I use pnpm firebase since few days to develop a firebase application.

Everything is running locally and I want now to deploy my app.

But I have an issue because firebase functions fail installing dependencies:
npm ERR! Unsupported URL Type "workspace:": workspace:^1.0.0
It is quite logical as workspace:^1.0.0 is only working with pnpm and also my monorepo dependency is only available locally...

I tried then to bundle my app locally but here starts the nightmare... I have a lot of unresolved dependencies! (I added more than 100 of them but all the time new unresolved dependencies appear.

Is there anybody facing the same issues?

Regards,

Jean-Guy

**Comment by rondered** [comment]
Facing the same issue. how did you solve it?

**Reply by BUONJG** [reply]
Hello Ron,

I'm switching automatically to standard npm in my CI:
- I publish each workspace packages to npm
- I update package.json to point to the last published version
- I run npm install
- I also optimized the process publishing packages only if a change occured because my CI was getting too slow...

It is not perfect but it is working. I spent like a day to make it work.

**Reply by shinaBR2** [reply]
I am facing the same problem, trying to follow [this answer](https://github.com/firebase/firebase-tools/issues/653#issuecomment-1312686210).
Unfortunately, It still does not work for me due to issues regarding copy dependencies. Anyway, you guys can give it a try.

**Reply by BUONJG** [reply]
Thank you very much for sharing this answer, I will have a look after holidays and keep you informed!
Regards,
Jean-Guy

**Reply by shinaBR2** [reply]
Hi, I did a hack to solve this problem: https://sworld-docs.netlify.app/blog/firebase-functions-deployment

**Comment by JohnXG** [comment]
I had the same issue when working on Glitch. Figured out a workaround, at least for now.

1 install pnpm at the monorepo root directory
2. run pnpm indirectly by calling npx first, e.g. npx pnpm install, for all scripts and terminal commands.

Here is my example, [https://glitch.com/edit/#!/deadpan-cerulean-chime](https://glitch.com/edit/#!/deadpan-cerulean-chime)

**Comment by bgrand-ch** [comment]
@BUONJG @shinaBR2 @JohnXG Hello, the solution I implemented:

https://stackoverflow.com/questions/75785161/how-to-use-pnpm-workspace-protocol-and-deploy-firebase-functions

**Reply by shinaBR2** [reply]
Thanks @bgrand-ch nice help

**Reply by bgrand-ch** [reply]
@rondered ℹ️
